replace MOM_stochastics_stub with stochastic_physics stub

pjpegion opened the pull request:
@jiandewang I made the code changes the follows my understanding of the reviewer's comments. I couldn't push directly to your branch so here is a PR.  
  I tested this using the data-atmosphere regression tests, and both the control and stochastic physics tests pass.

pjpegion committed:
replace MOM_stochastics_stub with stochastic_physics stub

pjpegion committed:
change eq to eqv

jiandewang commented:
@pjpegion Thanks for the updating.
I just made a quick test in UFS but compiling failed in MOM_energetic_PBL.F90. Did you forget to commit some changes or maybe there is something I missed here ?
you can see one of the error on HERA at /scratch1/NCEPDEV/stmp2/Jiande.Wang/FV3_RT/rt_15755/compile_004/err around line 550.
My ufs working dir: /scratch2/NCEPDEV/climate/Jiande.Wang/working/Phil-stochastic/1105/UFS-test, here I used your latest MOM6 to replaced the current one.
Can you show me your related dir so that I can make a comparison ?
 
Phil:
 The above failure is due to my bad, I forgot to modify mom6_files.cmake based on your code re-structure. Now my tests show datm-ocean compiling fine but S2S or S2SW still failed in compiling as stochastic_physics module is defined both in MOM6/config_src/external/stochastic_physics/ and stochastic_physics/stochastic_physics.F90, see error at
/scratch1/NCEPDEV/stmp2/Jiande.Wang/FV3_RT/rt_27927/compile_002/err

_MOM6-interface/libmom6.a(stochastic_physics.F90.o):/scratch2/NCEPDEV/climate/Jiande.Wang/working/Phil-stochastic/1105/UFS-test-1/MOM6-interf
ace/MOM6/config_src/external/stochastic_physics/stochastic_physics.F90:4: first defined here
stochastic_physics/libstochastic_physics.a(stochastic_physics.F90.o): In function `init_stochastic_physics':
/scratch2/NCEPDEV/climate/Jiande.Wang/working/Phil-stochastic/1105/UFS-test-1/stochastic_physics/stochastic_physics.F90:213: multiple defini
tion of `stochastic_physics_mp_init_stochastic_physics_ocn_'_

code and script: /scratch2/NCEPDEV/climate/Jiande.Wang/working/Phil-stochastic/1105/UFS-test-1

jiandewang commented:
update: for the total 10 of DATM jobs, 9 passed but datm_cdeps_stochy_gefs failed 
FATAL from PE     3: call to init_stochastic_physics_ocn failed
see err at /scratch1/NCEPDEV/stmp2/Jiande.Wang/FV3_RT/rt_31002/datm_cdeps_stochy_gefs

jiandewang commented:
can you share me your mom6_files.cmake ? Maybe I didn't modify it correctly since in your test stochastic run is OK

pjpegion commented:
I did have to modify the mom6_files.cmake and CMakeLists.txt to accommodate the changes in MOM6.  Please copy the files out of /scratch2/BMC/gsienkf/Philip.Pegion/ufs-weather-model.mom6pr/MOM6-interface and try again.    
 We will need to bring these changes into the ufs-weather-model when we merge these code changes back to EMC.

pjpegion committed:
add doxygen style comments

jiandewang commented:
I modified mom6_files.cmake correctly but forgot to modify CMakeLists.txt. 

For your test, if you go to the end of /scratch2/BMC/gsienkf/Philip.Pegion/ufs-weather-model.mom6pr/tests/log_hera.intel/run_datm_cdeps_stochy_gefs.log, it indicated the job failed.

For my new test, S2S failed in compiling, you can see log at 
/scratch1/NCEPDEV/stmp2/Jiande.Wang/FV3_RT/rt_4572/compile_002/err

pjpegion commented:
Hi Jiande,
  I did a test where I modified the mom6_files.cmake and CMakelists.txt to
use the stub to make sure that the control still works. The failure you are
seeing is intentional since the stochastics is using the stub.  In my
effort to revert the mom6_files.cmake back to its original form. I messed
up.  Please copy the mom6_files.cmake again.
Sorry about that,
Phil


On Mon, Nov 8, 2021 at 9:52 AM jiandewang ***@***.***> wrote:

> I modified mom6_files.cmake correctly but forgot to modify CMakeLists.txt.
>
> For your test, if you go to the end of
> /scratch2/BMC/gsienkf/Philip.Pegion/ufs-weather-model.mom6pr/tests/log_hera.intel/run_datm_cdeps_stochy_gefs.log,
> it indicated the job failed.
>
> For my new test, S2S failed in compiling, you can see log at
> /scratch1/NCEPDEV/stmp2/Jiande.Wang/FV3_RT/rt_4572/compile_002/err
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jiandewang/MOM6/pull/6#issuecomment-963359345>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJIRVJEK3MFXXWOS52EIFCTUK7547ANCNFSM5HOZXDDA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Phil Pegion (he/him/his)
Physical Scientist
NOAA/Physical Sciences Laboratory
(303) 497-7897
***@***.***

jiandewang commented:
you mean remove the following from mom6_files.cmake, right ?
MOM6/config_src/external/stochastic_physics/stochastic_physics.F90

pjpegion commented:
Yes

On Mon, Nov 8, 2021 at 10:08 AM jiandewang ***@***.***> wrote:

> you mean remove the following from mom6_files.cmake, right ?
> MOM6/config_src/external/stochastic_physics/stochastic_physics.F90
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jiandewang/MOM6/pull/6#issuecomment-963373997>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJIRVJGPRKCTQCMNWQEXT7DUK772DANCNFSM5HOZXDDA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Phil Pegion (he/him/his)
Physical Scientist
NOAA/Physical Sciences Laboratory
(303) 497-7897
***@***.***

jiandewang commented:
compiling fine now, let's wait a bit for the jobs to finish. Thanks

jiandewang commented:
jobs run fine, I am going to merge your PR. In the meantime can you write a sentence to describe the code structure you made in my PR to GFDL at https://github.com/NOAA-GFDL/MOM6/pull/1538  ?
